core/loadpoint.go

```diff
@@ -609,8 +612,30 @@ func (lp *LoadPoint) syncCharger() {
 
 // setLimit applies charger current limits and enables/disables accordingly
 func (lp *LoadPoint) setLimit(chargeCurrent float64, force bool) error {
+	// apply load management
+	forceCurrentChange := false
+	if lp.CircuitPtr != nil && chargeCurrent > 0.0 {
+		maxCur := lp.CircuitPtr.GetRemainingCurrent()
+		// maxCur includes the consumption of this loadpoint, so adjust using consumer interface
+		curAct, err := lp.GetCurrent()
+		if err != nil {
+			lp.log.ERROR.Printf("error getting current consumption: %s", err)
+			return err
+		}
+		newCur := maxCur + curAct
```

Comment:
Was passiert denn, wenn `GetCurrent` einen negativen Strom liefert? Der darf ja nicht addiert werden. Testcase?

Was passiert wenn mehrere child circuits negative Ströme liefern? Müssen die auch limitiert werden?

Reply:
Negative Ströme heisst negativer Verbrauch = Einspeisung. In einem Stromkreis kann auch ein Produzent sein. Standardfall ist der main Gridmeter, wo die PV einspeist. Generell kann in jedem Circuit ein Produzent sein, und auch ein Consumer könnte was einspeisen (V2G).
Lastmanagement soll sicherstellen, dass der Verbrauch nicht über einen Schwellwert in einem Stromkreis geht.

Reply:
Das ist mir schon klar. Und wie stellst Du das bei Addition sicher wenn auf L1 Bezug ist und die Einspeisung nur auf L2? M.E. ist die Addition hier falsch.

Reply:
Es wird ja immer größte aller 3 Ströme verwendet. Sollte p1 negativ und p2 positiv sein, gewinnt p2. 
Wir arbeiten nicht phasengenau aktuell.

Reply:
Und trotzdem ist das doch falsch. Bei +10A und -10A auf unterschiedlichen Phasen ist eine 6A Sicherung immer überlastet.

Reply:
Bei externer Überlastung kann evcc aber auch im Zweifelsfall nicht mehr tun als das Laden ganz zu unterbinden.

Reply:
Was ja dann die Aufgabe von Lastmanagement wäre?

Reply:
Lass uns das mal aufmalen und ggf. mündlich diskutieren, denn ich bin mir nicht sicher ob wir alle das gleiche Verständnis haben oder einfach nur aneinander vorbei denken ;)

Reply:
Lastmanagement würde im Fall +10A und -10A bei limit 6A den LP nicht freigeben (verfügbarer Strom = 0). Wie @premultiply sagt, kommt das dann von anderen Verbrauchern, welche EVCC nicht steuert. Das würde aber dann auch ohne EVCC und ohne Lastmanagement passieren.

Reply:
Was würde es bei Limit 10A machen? Anders gefragt: wie werden abweichende Vorzeichen behandelt? Ich hab nur konkret auf die Addition oben geschaut. Gibts evtl. einen Testcase für das Szenario? Hier nochmal eine Skizze was ich im Kopf hatte 
![85FE8077-B51F-4F48-BDFC-0D44C08D1BE8](https://user-images.githubusercontent.com/184815/205853630-91b77e3c-4255-495e-ab4d-0256ca515b8c.jpeg)

Reply:
Hier müsste noch das Netz eingezeichnet werden. Annahme: C1 ist das Grid, die WB ein Loadpoint. Dann wird der LP dem Circuit C1 zugewiesen. Wenn C1 ein eigenes Meter hat, dann ist der Strom ja klar. Wenn nicht, errechnet C1 seinen Strom aus der WB (LP). In dieser Konstellation (C1 = Grid) muss ein Gridmeter vorhanden sein, sonst hat es keinen Sinn. Lastmanagement nur für C3 mach keinen Sinn, wenn nur eine WB dran hängt.
Die rechte Seite verstehe ich nicht.
Lastmanagement ist dafür gedacht, an *einem* Punkt den Bezug zu begrenzen. Der Punkt ist eine Unterverteilung mit Zähler, oder mehrere LP welche zusammen einen virtuellen Stromkreis bilden.